README.md

```diff
@@ -1,49 +1,64 @@
 [![Build Status](https://travis-ci.org/kubernetes-csi/external-provisioner.svg?branch=master)](https://travis-ci.org/kubernetes-csi/external-provisioner)
 
-# Kubernetes external provisioner that works with CSI volumes
+# external-provisioner
 
-This is an example external provisioner for Kubernetes which provisions using CSI Volume drivers..  It's under heavy development, so at this time README.md is notes for the developers coding.  Once complete this will change to something user friendly.
+The external-provisioner is a sidecar container that watches the Kubernetes API
+server for PersistentVolumeClaim objects for the CSI driver.
 
-# Build
+It calls `CreateVolume` against the specified CSI endpoint to provision a new
+volume, and `DeleteVolume` to delete the volume.
 
-```bash
-make csi-provisioner
-```
+## Feature Status
+Overall: GA
 
-# Test
+* Topology: Beta
+* Snapshots: Alpha
 
-## Start Kubernetes
+## Compatibility
 
-Run a local kubernetes cluster built from latest master code
+This information reflects the head of this branch.
 
-## Run Storage Provider
+| Compatible with CSI Version                                                                | Container Image                 | Min K8s Version |
+| ------------------------------------------------------------------------------------------ | ------------------------------- | --------------- |
+| [CSI Spec v1.0.0](https://github.com/container-storage-interface/spec/releases/tag/v1.0.0) | quay.io/k8scsi/csi-provisioner  | 1.14            |
 
-### Use HostPath drivers
 
-Go to drivers and run:
+## Usage
 
-```bash
-bin/hostpathplugin --drivername mydriver  --endpoint unix://tmp/csi.sock --nodeid foobar -v=5
-```
+### Required arguments
 
-### Start external provisioner
+* `--csi-address`: This is the path to the CSI driver UNIX domain socket inside
+  the pod that the `external-provisioner` container will use to issue CSI
+  operations (e.g. `/csi/csi.sock`).
 
-```bash
-bin/csi-provisioner --kubeconfig /var/run/kubernetes/admin.kubeconfig --alsologtostderr
-```
+### Common arguments
 
-### Create Storage class, PVC, and Pod
+* `--timeout`: Number of nanoseconds before a CreateVolume or DeleteVolume operation
```

Comment:
@jsafrane I noticed the value is in nanoseconds. Is that what we want or would seconds be more reasonable?

Reply:
It's `flag.Duration`, so it accepts suffixes like `s` for seconds. The description is wrong / misleading.

https://github.com/kubernetes-csi/external-provisioner/blob/e9387dc7f8079dd745c10dca185257dcf2994abb/cmd/csi-provisioner/csi-provisioner.go#L61

Reply:
Btw, it would be probably better to rework whole README in a separate PR. I'd like to have there the same format / information as in attacher (https://github.com/kubernetes-csi/external-attacher/pull/130).